add delete button and corresponding backbone destroy functionality

majornorth opened the pull request.

majornorth commented:
@ezarowny +1? :+1:

ezarowny commented on utils/static/utils/scripts/app/views.js:
```diff
@@ -20,6 +22,10 @@ define(['jquery', 'underscore', 'backbone', 'marionette', 'app/models'], functio
             this.model.setDone(isDone);
+        },
+
+        deleteMofo: function() {
```
deleteTodo perhaps?

ezarowny commented on utils/templates/utils/index.html:
```diff
@@ -15,7 +15,7 @@
     <script type="text/html" id="list-item">
         <input class="item-done" type="checkbox"></input>
-        <%= title %>
+        <%= title %><button class="delete">Delete!</button>
```
Move button onto next line for readability's sake?

Commit:
Add improved functionality and styling

- Update method for toggling isDone
- Add support for 'Enter' on keypress
- Add toggle for completed class
- Add strikethrough style when completed
- Put style inside css folder

majornorth commented:
@ezarowny updated based on your feedback. Mergeruskie!

ezarowny commented:
Squash it and we're good to go!

Commit:
change toggle to add and remove
